packages/compiler/src/compiler/readMetadata.ts

```diff
@@ -419,13 +425,20 @@ export class ReadMetadata {
             }
           })
           this.accumulatedToolCalls = refReadMetadata.accumulatedToolCalls
-
-          this.referencedPrompts[refPromptPath] = refPromptMetadata.hash
+          resolvedRefPrompt = refReadMetadata.resolvedPrompt
         } catch (error: unknown) {
-          if (error instanceof CompileError) return
           this.baseNodeError(errors.referenceError(error), node)
-          return
+          resolvedRefPrompt = `/* <${REFERENCE_PROMPT_TAG} ${REFERENCE_PROMPT_ATTR}="${refPromptPath}" /> */`
         }
+
+        /* Replace the reference tag with the actual referenced prompt */
+        const start = node.start! + this.resolvedPromptOffset
+        const end = node.end! + this.resolvedPromptOffset
+        const pretext = this.resolvedPrompt.slice(0, start)
+        const posttext = this.resolvedPrompt.slice(end)
+        this.resolvedPrompt = pretext + resolvedRefPrompt + posttext
+        this.resolvedPromptOffset += resolvedRefPrompt.length - (end - start)
```

Comment:
This could be a nice method with a nice name

Reply:
agree. Will add it to a next pr